src/connectionsetup.ts

```diff
@@ -0,0 +1,153 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2022 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import { ConfigurationTarget } from 'vscode';
+import { Commands } from './commands';
+import { ResourceResolver } from './webview';
+import * as util from './util';
+
+let connectionSetupPanel: vscode.WebviewPanel;
+
+export function connectToSonarQube(context: vscode.ExtensionContext) {
+  return () => {
+    lazyCreateConnectionSetupPanel(context);
+    connectionSetupPanel.webview.html = renderConnectionSetupPanel(context, connectionSetupPanel.webview);
+    connectionSetupPanel.webview.onDidReceiveMessage(handleMessage);
+    connectionSetupPanel.iconPath = {
+      light: util.resolveExtensionFile('images/sonarqube.svg'),
+      dark: util.resolveExtensionFile('images/sonarqube.svg')
+    };
+    connectionSetupPanel.reveal();
+  };
+}
+
+function lazyCreateConnectionSetupPanel(context: vscode.ExtensionContext) {
+  if (!connectionSetupPanel) {
+    connectionSetupPanel = vscode.window.createWebviewPanel(
+      'sonarlint.ConnectionSetup',
+      'SonarQube Connection',
+      vscode.ViewColumn.Active,
+      {
+        enableScripts: true
+      }
+    );
+    connectionSetupPanel.onDidDispose(
+      () => {
+        connectionSetupPanel = undefined;
+      },
+      null,
+      context.subscriptions
+    );
+  }
+}
+
+function renderConnectionSetupPanel(context: vscode.ExtensionContext, webview: vscode.Webview) {
+  const resolver = new ResourceResolver(context, webview);
+  const styleSrc = resolver.resolve('styles', 'connectionsetup.css');
+  const toolkitUri = resolver.resolve(
+    'node_modules',
+    '@vscode',
+    'webview-ui-toolkit',
+    'dist',
+    'toolkit.js'
+  );
+  const webviewMainUri = resolver.resolve('webview-ui', 'connectionsetup.js');
+
+  return `<!doctype html><html lang="en">
```

Comment:
just an idea, can/should this html snippet be moved to a separate file?

Reply:
It could, as long as it remains static; my idea is to make it dynamic (soon) to support SonarCloud, and reuse it for the modification of a connection.